### PR TITLE
add create update delete to sync api

### PR DESCRIPTION
**Pull request by kti-sam**

### CODE REVIEW

File: djpsa/halo/sync.py [hunk 1]
```diff
@@ -41,13 +41,42 @@ def _unpack_records(self, response):
         return records
 
 
+class CreateMixin:
+
+    def create(self, data, *args, **kwargs):
+        body = self._convert_fields_to_api_format(data)
```
Comment (kti-matt) [inline djpsa/halo/sync.py:47]: We've done this a lot in the past but PyCharm doesn't like it and it's right- it makes things more confusing than necessary.

PyCharm doesn't like the unresolved attribute reference:

image

`_convert_fields_to_api_format` should go on a class that `CreateMixin` and `UpdateMixin` inherit from, and not on `HaloSynchronizer`.

Comment (kti-matt) [reply djpsa/halo/sync.py:47]: And the unresolved reference to `client` should be handled simply by adding a `client = None` to the parent class.

File: djpsa/halo/sync.py [hunk 2]
```diff
@@ -41,13 +41,42 @@ def _unpack_records(self, response):
...
+        return self.client.create(body)
+
+
+class UpdateMixin:
+
+    def update(self, record_id, data, *args, **kwargs):
```
Comment (kti-matt) [inline djpsa/halo/sync.py:53]: As discussed- make this update the model using update_or_create. Probably it should take the model as an argument, not an integer ID.

Probably it should not save the model itself, but let the caller save it?

Comment (kti-sam) [reply djpsa/halo/sync.py:53]: No, it shouldn't give the caller a chance to not save it. You would always want it saved as its already been updated in the remote PSA. Not saving would just de-sync the record.

**Pull request merged**